aws/table_aws_directory_service_directory.go

```diff
@@ -383,6 +389,12 @@ func getDirectoryServiceEventTopics(ctx context.Context, d *plugin.QueryData, h
 func getDirectoryServiceSnapshotLimit(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
 	directory := h.Item.(types.DirectoryDescription)
 
+	// This operation is not supported for for Shared MicrosoftAD directories
+	// Error: aws: operation error Directory Service: GetSnapshotLimits, https response error StatusCode: 400, ClientException: Snapshot limits can be fetched only for VPC or Microsoft AD directories.
+	if directory.Type == "SharedMicrosoftAD" {
```

Comment:
@ParthaI, in this case, the error did not say that it is not supported for `SharedMicrosoftAD`.

Reply:
@bigdatasourav, the error message says `Snapshot limits can be fetched only for VPC or Microsoft AD directories`
but the type value we are getting `SharedMicrosoftAD` means it is a shared directory. If it is not a shared directory the type would be `MicrosoftAD`.